### PR TITLE
Add SubscribeToRotatedIdentities interface 

### PR DESCRIPTION
**Pull request by meyskens**
This change adds a SubscribeToRotatedIdentities function.
This gives a channel which is used to pass identity updates
back from the certificate proider to the auth manager.
In the auth manager there can better be acted upon to
receive the IDs and re-trigger a mTLS handshake if needed.

```release-note
Add a mechanism for the SPIRE server to signal rotated certificates for re-authenticating connections
```

### CODE REVIEW

File: pkg/auth/spire/certificate_provider.go
Comment (youngnick) [inline pkg/auth/spire/certificate_provider.go]: Given the discussion we had about maybe changing this, should we make this a const somewhere? I think it's in more than just here now?

**Review by youngnick — APPROVED** [review]
LGTM with one non-blocking comment about `cilium-id`.

File: Documentation/cmdref/cilium-agent.md
Comment (mhofstetter) [inline Documentation/cmdref/cilium-agent.md]: shouldn't we name this `--mesh-auth-rotated-identities-queue-size` as mentioned in the comment -> with "identities"?

File: pkg/auth/manager.go
Comment (mhofstetter) [inline pkg/auth/manager.go:34]: tempted to say we should kind of "duplicate" the type `CertificateRotationEvent` within the authhandler - and convert at the boundary. but since we're also using other types it seems ok.

File: pkg/auth/spire/delegate.go
Comment (mhofstetter) [inline pkg/auth/spire/delegate.go]: might be worth to use the loglimiter to prevent flooding the log.

-> https://github.com/cilium/cilium/blob/master/pkg/auth/monitor/monitor.go#L72-L75

**Review by mhofstetter — CHANGES_REQUESTED** [review]
looks good, some minor suggestions

File: pkg/auth/spire/certificate_provider.go
Comment (meyskens) [reply pkg/auth/spire/certificate_provider.go]: It is in a few places once @sayboras's work on creating the IDs merged we can do a replace all of them to have it all done at once

File: pkg/auth/manager.go
Comment (meyskens) [reply pkg/auth/manager.go:34]: we already have a shared struct structure here, i wouldn't like to do 1 to 1 conversion into types that do the same unless really necessary...

File: pkg/auth/spire/delegate.go
Comment (meyskens) [reply pkg/auth/spire/delegate.go]: Good tip!

**Review by mhofstetter — APPROVED** [review]

**Comment by julianwiedmann** [comment]
Let's wait for tests to complete :hourglass_flowing_sand:. mlh will set the label automatically if all is green.

**Comment by youngnick** [comment]
Seems like this is currently blocked by #24471.

File: pkg/auth/spire/certificate_provider.go
Comment (youngnick) [reply pkg/auth/spire/certificate_provider.go]: I think @sayboras has already changed this to `identity` in his PR, so we might need to update here as well, maybe wait until the other PR is merged?

Comment (sayboras) [reply pkg/auth/spire/certificate_provider.go]: yup, I have cleaned this up in #24765

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

**Comment by sayboras** [comment]
integration test failures seem related to this change

```
--- FAIL: TestSpireDelegateClient_GetCertificateForIdentity (0.01s)
                                                                   
    --- FAIL: TestSpireDelegateClient_GetCertificateForIdentity/get_certificate_for_numeric_identity (0.00s)
                                                                                                            
    certificate_provider_test.go:391: SpireDelegateClient.GetCertificateForIdentity() error = no SPIFFE ID for spiffe://test.cilium.io/identity/1234, wantErr false
────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                
--- FAIL: TestSpireDelegateClient_SubscribeToRotatedIdentities (0.00s)
                                                                      
    --- FAIL: TestSpireDelegateClient_SubscribeToRotatedIdentities/receive_1_updated_event (0.00s)
                                                                                                  
level=error msg="failed to convert spiffe ID spiffe://test.cilium.io/cilium-id/1234 to numeric identity" error="SPIFFE ID spiffe://test.cilium.io/cilium-id/1234 does not belong to our trust domain or is not in the valid format" subsys=spire
    certificate_provider_test.go:486: SpireDelegateClient.SubscribeToRotatedIdentities() = [], want [{1234}]
────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                
--- FAIL: TestSpireDelegateClient_ValidateIdentity (0.00s)
                                                          
    --- FAIL: TestSpireDelegateClient_ValidateIdentity/validate_with_correct_SPIFFE_ID (0.00s)
                                                                                              
    certificate_provider_test.go:219: SpireDelegateClient.ValidateIdentity() = false, want true
────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                
--- FAIL: TestSpireDelegateClient_sniToSPIFFEID (0.00s)
                                                       
    --- FAIL: TestSpireDelegateClient_sniToSPIFFEID/convert_valid_numeric_identity (0.00s)
 ```

**Comment by meyskens** [comment]
Rebased with fixes to test

**Comment by meyskens** [comment]
/re-test test-runtime

due to network issue
```
docker: Error response from daemon: Get "[https://registry-1.docker.io/v2/":](https://registry-1.docker.io/v2/) read tcp 10.0.2.15:33340->52.1.184.176:443: read: connection reset by peer.
```

**Pull request merged**
